docs/src/main/sphinx/connector/hive.rst

```diff
@@ -688,24 +688,46 @@ ALTER TABLE EXECUTE
 ^^^^^^^^^^^^^^^^^^^
 
 The connector supports the following commands for use with
-:ref:`ALTER TABLE EXECUTE <alter-table-execute>`:
+:ref:`ALTER TABLE EXECUTE <alter-table-execute>`.
 
-* ``optimize``: collapse files in a non-transactional table up to a threshold
-  defined in the ``file_size_threshold`` parameter. For example, the following
-  statement collapses files in a table that are under 10 megabytes in size:
+optimize
+~~~~~~~~
 
-  .. code-block:: sql
+The ``optimize`` command is used for rewriting the content
+of the specified non-transactional table so that it is merged
+into potentially fewer larger files.
```

Comment:
`into potentially fewer larger files` -> `into fewer but larger files`

Reply:
If there is only one file to "optimize", the file will be rewritten. The outcome of of the operation will be to have the same amount of files (not fewer) with the same size.
I would argue that `into potentially fewer larger files` fits to the functionality currently provided by `OPTIMIZE`

Reply:
I agree that mathematically having "potentally" in the sentence is more accurate. But without further explanation, it is not providing any useful information to the reader. The intent of the method is to create fewer but bigger files and IMO that is enough for main documentation part.

If we want to document caveats though, I would add a separate section saying that sometimes procedure may not work as expected, and explain the reason, and how to deal with that.